### DISCUSSION TITLE
OTel collector distribution components

### DISCUSSION THREAD

**Discussion by ophiryael** [teletrace/teletrace#342] [General]
When building an OpenTelemetry collector distribution, we have to choose which components (receivers/processors/extensions/exporters) should be included in the build.
You can read more about the types of distributions (Pure/Plus/Minus) in the [official docs](https://opentelemetry.io/docs/concepts/distributions/#what-is-a-distribution).

The question is whether we want to bundle only the necessary components, which are the OTLP receiver, batch processor and the custom Elasticsearch exporter, or do we also want to bundle unrelated components (e.g. Prometheus exporter).

The benefit for the user when bundling more components, is that instead of setting up two collectors - the official OTel collector and our own distribution, he will be able to choose to use only our distribution for all of his OTel requirements (e.g. logs, metrics, and maybe other trace exporters), 
The downside of bundling more components is the increased size of the collector (which affects related areas such as CI/CD build time) and keeping components up to date with the core/contrib collectors.

@haimrait @maorlx @tvaintrob @galbash @ronyis @itaykat @roybh1

**Comment by tvaintrob** [comment]
IMO we should only bundle required components that are needed for operation, but we should include stuff like Prometheus exporter to export metrics on the collector itself

**Reply by ophiryael** [reply]
Not sure I follow, you said that we should only bundle the required components and then said that we should also bundle the Prometheus exporter.

**Reply by tvaintrob** [reply]
i meant only essential plugins for operation, only the ones used in the pipeline, but also include stuff to operate the service itself, like Prometheus metrics exporter, the gist is to avoid bundling unused exporters / receivers like jaeger-receiver for example

**Comment by roybh1** [comment]
From the users' perspective, this is the the Lupa Backend component, not an actual otel collector. I think we should only add the essential components.

**Comment by maorlx** [comment]
I think we should bundle only the necessary components. IMO it makes more sense to treat our otel only as the lupa backend component (and not the actual user OTEL collector).
IMO the prometheus exporter isn't considered 'required' as the OTEL Collector should expose its own metrics and the user can decide whether to scrap metrics from it
